app/models/gather_repository_maintenance_stats.rb

```diff
@@ -0,0 +1,75 @@
+class GatherRepositoryMaintenanceStats
+    def self.gather_stats(repository: nil)
+        repository = Repository.first if repository.nil?
+        client = AuthToken.v4_client
+        v3_client = AuthToken.client
```

Comment:
That's a good point. I think the way `AuthToken` works is that is verifies the client has 500 requests remaining at least, so I don't think it will be an issue in here.